MobileWallet/UIElements/AnimatedRefreshingView.swift

```diff
@@ -80,27 +85,42 @@ private class RefreshingInnerView: UIView {
         spinner.hidesWhenStopped = true
 
         switch type {
-            case .loading:
-                emojiLabel.text = "⏳"
-                spinner.startAnimating()
-                statusLabel.text = NSLocalizedString("Checking for Updates", comment: "Refresh view")
-                statusLabel.textColor = Theme.shared.colors.refreshViewLabelLoading
-            case .receiving:
-                emojiLabel.text = "🤝"
-                spinner.startAnimating()
-                statusLabel.text = NSLocalizedString("Receiving new transaction", comment: "Refresh view")
-                statusLabel.textColor = Theme.shared.colors.refreshViewLabelLoading
-            case .success:
-                statusLabel.text = NSLocalizedString("You are up to date!", comment: "Refresh view")
-                spinner.stopAnimating()
-                statusLabel.textColor = Theme.shared.colors.refreshViewLabelSuccess
+        case .loading:
+            emojiLabel.text = "⏳"
+            spinner.startAnimating()
+            statusLabel.text = NSLocalizedString("Checking for Updates", comment: "Refresh view")
+            statusLabel.textColor = Theme.shared.colors.refreshViewLabelLoading
+        case .receiving:
+            emojiLabel.text = "🤝"
+            spinner.startAnimating()
+            statusLabel.text = NSLocalizedString("Receiving new transaction", comment: "Refresh view")
+            statusLabel.textColor = Theme.shared.colors.refreshViewLabelLoading
+        case .success:
+            statusLabel.text = NSLocalizedString("You are up to date!", comment: "Refresh view")
+            spinner.stopAnimating()
+            statusLabel.textColor = Theme.shared.colors.refreshViewLabelSuccess
+        case .txWaitingForRecipient:
+            emojiLabel.text = ""
+            spinner.stopAnimating()
+            statusLabel.text = NSLocalizedString("Waiting for recipient to accept transaction", comment: "Refresh view")
+            statusLabel.textColor = Theme.shared.colors.refreshViewLabelLoading
+        case .txWaitingForSender:
```

Comment:
Some of these types look to me like they're only for the tx details screen, so I'm not sure if it's the best location.